### ISSUE TITLE
How to get back to the previous PDF viewer?

### ISSUE THREAD

**Issue by carterbao** [brookhong/Surfingkeys#1594]
In the most recent release (Dec. 29th, 2021), the PDF viewer has been changed to a newer version. However, it brought some inconveniences compared to the older version:
1. There are no zoom in/out buttons and the page cannot be easily adapted to a comfortable size.
2. The page is left aligned by default which is not so good-looking, and I can't find a way to change the alignment.
3. No download/rotate buttons, and no page numbers as well.

Also, there seems to be some bugs:
1. When pressing "v" to enter view mode, there are no letters showing up on every block for the user to select the position of cursor.

My platform: Chrome 96.0.4664.55, Mac OS X

The question is that, is there a way to switch back to the previous PDF viewer? Or would you fix these issues in the future?
Thanks a lot!

**Comment by yusefmosiah** [comment]
I'm having the same issue. Although for me it seems that it's defaulted back to the stock Chrome PDF viewer.

**Comment by 0900304132** [comment]
`;s`

**Comment by 0900304132** [comment]
key ';' + key 's'

**Comment by yusefmosiah** [comment]
No, that isn't working. ;s displays a brief modal pop up that the pdf viewer was enabled/disabled, but regardless of its status, it defaults back to the stock pdf viewer.

**Comment by carterbao** [comment]
> key ';' + key 's'

yeah I think `;s` switches to the default PDF viewer of chrome. However, I'd prefer the previous PDF viewer provided by Surfingkeys as it supported vim-like shortcuts.

**Comment by RedBotz** [comment]
I'll echo this - the update limit a piece of functionality that I really appreciated and used often.  Hoping it can be brought back!

**Comment by zqni** [comment]
Seems I couldn't use any shortcuts in the current PDF viewer, sad about that. Chrome  97.0.4692.71, Win11 and Surfingkeys1.0.2

**Comment by appollo101** [comment]
## Error details



SurfingKeys: 1.0.3

Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/97.0.4692.71 Safari/537.36

URL: every pdf file

## Context

**Please replace this with a description of how you were using SurfingKeys.**

This version of pdfviewer puts all texts in the first page, overlapping with each other, and displays no page break. PDF files look like images instead of PDFs.

**Comment by gvteja** [comment]
+1. I really appreciate using SurfingKeys pdf editor, which allowed me to navigate using SurfingKeys's navigation.

**Comment by zqni** [comment]
The pdf viewer of the current edition seems to work with all shortcuts. But it compresses all pdf files as a long image which means people couldn't even download the file opened in the Surfingkeys extension.

**Comment by navakelvin** [comment]
So the pdf.js viewer has disappeared?

**Comment by Sbozzolo** [comment]
@brookhong 

Could you please comment on this issue? For example, why was the pdf viewer changed?

**Comment by brookhong** [comment]
The previous PDF viewer was actually copied from [mozilla/pdf.js](https://github.com/mozilla/pdf.js/blob/master/web/viewer.html), the build process is sort of hack. Since the refactor for 1.0, the hack steps of copying implementation from [mozilla/pdf.js](https://github.com/mozilla/pdf.js/blob/master/web/viewer.html) was ditched.

And I'm planning to remove pdf viewer from Surfingkeys, as the original thoughts to embed a pdf viewer is just to enable Surfingkeys on a tab for pdf viewer, which seems not be a common need for most users, and embedding a pdf viewer in Surfingkeys bloats the extension itself.

For those needs Surfingkeys on every tab(including pdf viewer), my chromium build could be an option.

**Comment by iamashwin99** [comment]
can we instead of completely removing it, split it into two extensions? One that wraps the pdf in pdf.js and the other (this one) that just checks for the other extension and based on that use one or the other viewer.
That way you don't bulk up the current extension.

**Comment by brookhong** [comment]
I changed my mind and brought this back in 1.17.7, closing this.
